com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,225 @@
+                asBanner()?.visibility = visibility
+            }
+        }
+    }
+
+    fun clearLoaded(): Boolean {
+        if (ad is HeliumFullscreenAd)
+            return asFullScreen()?.clearLoaded() ?: false
+        if (ad is HeliumBannerAd) {
+            HeliumUnityBridge.runTaskOnUiThread { asBanner()?.clearAd() }
+            return true
+        }
+        return false
+    }
+
+    fun readyToShow(): Boolean {
+        return if (!startedLoad) false else try {
+            when (ad) {
+                is HeliumFullscreenAd -> asFullScreen()?.readyToShow() ?: false
+                is HeliumBannerAd -> {
+                    Log.w(TAG, "This should never be called, banners do not have readyToShow")
+                    false
+                }
+                else -> false
+            }
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
+            false
+        }
+    }
+
+    fun destroy() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            destroyBannerLayout()
+            ad.destroy()
+        }
+    }
+
+    private fun createBannerLayout(screenLocation: Int) {
+        if (ad !is HeliumBannerAd)
+            return
+
+        bannerLayout?.let {
+            it.removeAllViews()
+        }
+
+        // Create the banner layout on the given position.
+        // Check if there is an already existing banner layout. If so, remove it. Otherwise,
+        // create a new one.
+        bannerLayout?.let {
+            val bannerParent = it.parent as FrameLayout
+            bannerParent.removeView(it)
+        } ?: run {
+            bannerLayout = RelativeLayout(activity)
+            bannerLayout?.setBackgroundColor(Color.TRANSPARENT)
+        }
+
+        /*
+            //     TopLeft = 0,
+            //     TopCenter = 1,
+            //     TopRight = 2,
+            //     Center = 3,
+            //     BottomLeft = 4,
+            //     BottomCenter = 5,
+            //     BottomRight = 6
+        */
+        var bannerGravityPosition = 0
+        when (screenLocation) {
+            0 -> bannerGravityPosition = Gravity.TOP or Gravity.LEFT
+            1 -> bannerGravityPosition = Gravity.TOP or Gravity.CENTER_HORIZONTAL
+            2 -> bannerGravityPosition = Gravity.TOP or Gravity.RIGHT
+            3 -> bannerGravityPosition = Gravity.CENTER_VERTICAL or Gravity.CENTER_HORIZONTAL
+            4 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.LEFT
+            5 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.CENTER_HORIZONTAL
+            6 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.RIGHT
+        }
+        bannerLayout?.gravity = bannerGravityPosition
+
+        // Attach the banner layout to the activity.
+        val pixels = displayDensity
+        try {
+            val bannerAd = asBanner()
+            when (if (bannerAd?.getSize() != null) bannerAd.getSize() else HeliumBannerSize.STANDARD) {
```

Comment:
can just use ad as it's been smartcast already
Be careful when using `?` as you have to handle the null case, too. it's easier to try to make things non null than try to handle every null case sometimes. Like crashing with TODO is probably fine here, but doesn't give the best error.